packages/astro/src/core/config/schemas/base.ts

```diff
@@ -487,4 +497,6 @@ export const AstroConfigSchema = z.object({
 		.default({}),
 });
 
+export type CspAlgorithm = z.infer<typeof cspAlgorithmSchema>;
```

Comment:
We usually don't export this kind of type from here. I think both `cspAlgorithmSchema` and `CspAlgorithm` should live in eg. `csp/config.ts`